### ISSUE TITLE
ArrayIndexOutOfBoundsException

### ISSUE THREAD

**Issue by hurzl** [woheller69/weather#17]
new location ->

```03-29 17:13:53.400 24221 24221 E AndroidRuntime: Process: org.woheller69.weather, PID: 24221
03-29 17:13:53.400 24221 24221 E AndroidRuntime: java.lang.ArrayIndexOutOfBoundsException: length=1; index=9
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at org.woheller69.weather.ui.RecycleList.WeekWeatherAdapter.onBindViewHolder(WeekWeatherAdapter.java:49)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at org.woheller69.weather.ui.RecycleList.WeekWeatherAdapter.onBindViewHolder(WeekWeatherAdapter.java:25)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:7065)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:7107)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView$Recycler.tryBindViewHolderByDeadline(RecyclerView.java:6012)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6279)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6118)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6114)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2303)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1627)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1587)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:665)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4134)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView.onMeasure(RecyclerView.java:3540)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.view.View.measure(View.java:25086)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6872)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1552)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.widget.LinearLayout.measureVertical(LinearLayout.java:842)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.view.View.measure(View.java:25086)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6872)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.cardview.widget.CardView.onMeasure(CardView.java:260)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.view.View.measure(View.java:25086)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView$LayoutManager.measureChildWithMargins(RecyclerView.java:9384)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1653)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1587)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:665)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4134)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView.dispatchLayout(RecyclerView.java:3851)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView.onLayout(RecyclerView.java:4404)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.view.View.layout(View.java:22466)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6304)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.view.View.layout(View.java:22466)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6304)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.viewpager.widget.ViewPager.onLayout(ViewPager.java:1775)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.view.View.layout(View.java:22466)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6304)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.view.View.layout(View.java:22466)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6304)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.coordinatorlayout.widget.CoordinatorLayout.layoutChild(CoordinatorLayout.java:1213)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayoutChild(CoordinatorLayout.java:899)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:919)
03-29 17:13:53.400 24221 24221 E AndroidRuntime: 	at android.view.View.layout(View.java:22466)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6304)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at androidx.drawerlayout.widget.DrawerLayout.onLayout(DrawerLayout.java:1231)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.view.View.layout(View.java:22466)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6304)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.view.View.layout(View.java:22466)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6304)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.view.View.layout(View.java:22466)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6304)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.view.View.layout(View.java:22466)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6304)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.view.View.layout(View.java:22466)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6304)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at com.android.internal.policy.DecorView.onLayout(DecorView.java:779)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.view.View.layout(View.java:22466)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6304)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:3157)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2614)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1745)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7768)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:967)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.view.Choreographer.doCallbacks(Choreographer.java:791)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.view.Choreographer.doFrame(Choreographer.java:726)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:952)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:883)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:100)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:214)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7356)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
03-29 17:13:53.401 24221 24221 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:940)
```

**Comment by woheller69** [comment]
which version are you using? I had this bug just before releasing V2.1.  Therefore I reverted this commit
https://github.com/woheller69/weather/commit/47ea66f98af573aede7f58fe18ee157167b661dd
and got rid of the problem.

**Comment by hurzl** [comment]
ok, I'm before 2.1, but index 8 is still > length=1

**Comment by woheller69** [comment]
Which version do you use then? 2.0?

**Comment by hurzl** [comment]
actually it's 2.1, sorry

**Comment by woheller69** [comment]
Does the problem go away if you reinstall V2.0?
What I changed between V2.0 and V2.1 is this https://github.com/woheller69/weather/commit/26028145afc355fbb53db1609d205d23766df103
Instead of naming the arrays today, tomorrow, etc I put them in a 2D array and filled them in a loop. But essentially that should be the same as before. I did not initialize the empty row at the end with 0 as before, but this should be done by the compiler anyway.

And also I added an onclick feature that scrolls course of day when you click on a week day.

Just to be sure F-Droid did not build something wrong, can you please try this? This is my latest version with
Photon Search API integrated (if activated in settings)
[RadarWeather3.0Beta.apk.zip](https://github.com/woheller69/weather/files/6223186/RadarWeather3.0Beta.apk.zip)

**Comment by hurzl** [comment]
no I can't work with the source atm.

**Comment by woheller69** [comment]
it is not the source, just a zipped apk. Cannot upload apk directly

**Comment by woheller69** [comment]
not even zipped. just rename it

**Comment by hurzl** [comment]
It wouldn't proof anything because it only occured once anyway

**Comment by woheller69** [comment]
ok, great. I love those bugs...

**Comment by woheller69** [comment]
The crash happened in WeekWeatherAdapter line 49:

         setIcon((int) dayValues[9], holder.weather);

Obviously dayValues was not yet filled for whatever reason. I will add a return if dayValues does not have the required length.

        public void onBindViewHolder(WeekForecastViewHolder holder, int position) {
             float[] dayValues = forecastData[position];
             if (dayValues.length!=11) return;

             setIcon((int) dayValues[9], holder.weather);
 
I can produce the bug if I reactivate the above mentioned reverted commit. Somehow this increases the probability for this bug (no idea why). With the "return" I can fix this as well.

**Comment by woheller69** [comment]
Fixed in V3.0
